semana17/aula1/Food4U/src/services/crypt/bcryptimplementation.ts

```diff
@@ -0,0 +1,16 @@
+import * as bcrypt from "bcrypt"
+import {CryptographyGateway} from "../../business/gateways/auth/CryptographyGateway";
+
+export class Bcryptimplementation implements CryptographyGateway {
+  private static BCRYPT_SALT_ROUNDS = 10;
+  async encrypt(word: string): Promise<string> {
+    const salt = await bcrypt.genSalt(Bcryptimplementation.BCRYPT_SALT_ROUNDS);
+    return await bcrypt.hash(
+        word,
+        salt
+    );
+  }
+  async compare(word:string, hash: string): Promise<boolean> {
+    return await bcrypt.compare(word, hash)
```

Comment:
Uma dica sobre um pequeno detalhe. Na maioria das vezes, dar um await em um return não possui nenhum efeito, e, portanto, ele pode ser omitido. Essa pergunta no StackOverflow ilustra bem isso: https://stackoverflow.com/questions/38708550/difference-between-return-await-promise-and-return-promise